src/ui/ui_review_menu_bagl.c

```diff
@@ -0,0 +1,859 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+#ifdef HAVE_BAGL
+#include <stdbool.h>
+#include <sys/types.h>
+#include <string.h>
+#include <stdint.h> 
+
+#include "ux.h"
+#include "os_io_seproxyhal.h"
+#include "ui_globals.h"
+#include "ui_review_menu.h"
+
+//////////////////////////////////////////////////////////////////////
+UX_STEP_NOCB(ux_approval_tx_data_warning_step,
+    pnn,
+    {
+      &C_icon_warning,
+      "Data",
+      "Present",
+    });
+
+UX_STEP_NOCB(
+    ux_approval_from_address_step,
+    bnnn_paging,
+    {
+      .title = "From Address",
+      .text = fromAddress
+    });
+
+UX_STEP_VALID(
+    ux_approval_confirm_step,
+    pbb,
+    io_seproxyhal_touch_tx_ok(NULL),
+    {
+      &C_icon_validate_14,
+      "Sign",
+      "transaction",
+    });
+
+UX_STEP_VALID(
+    ux_approval_reject_step,
+    pbb,
+    io_seproxyhal_touch_cancel(NULL),
+    {
+      &C_icon_crossmark,
+      "Cancel",
+      "signature",
+    });
+
+//////////////////////////////////////////////////////////////////////
+UX_STEP_NOCB(
+    ux_display_public_flow_1_step,
+    pnn,
+    {
+      &C_icon_eye,
+      "Verify",
+      "Address",
+    });
+UX_STEP_NOCB(
+    ux_display_public_flow_2_step,
+    bnnn_paging,
+    {
+      .title = "Address",
+      .text = toAddress,
+    });
+UX_STEP_VALID(
+    ux_display_public_flow_3_step,
+    pb,
+    io_seproxyhal_touch_address_ok(NULL),
+    {
+      &C_icon_validate_14,
+      "Approve",
+    });
+
+UX_DEF(ux_display_public_flow,
+  &ux_display_public_flow_1_step,
+  &ux_display_public_flow_2_step,
+  &ux_display_public_flow_3_step,
+  &ux_approval_reject_step
+);
+
+// Simple Transaction:
+//////////////////////////////////////////////////////////////////////
+UX_STEP_NOCB(
+    ux_approval_st_flow_1_step,
+    pnn,
+    {
+      &C_icon_eye,
+      "Verify",
+      fullContract
+    });
+UX_STEP_NOCB(
+    ux_approval_st_flow_2_step,
+    bnnn_paging,
+    {
+      .title = "Hash",
+      .text = fullHash
+    });
+
+UX_DEF(ux_approval_st_flow,
+  &ux_approval_st_flow_1_step,
+  &ux_approval_st_flow_2_step,
+  &ux_approval_from_address_step,
+  &ux_approval_confirm_step,
+  &ux_approval_reject_step
+);
+
+UX_DEF(ux_approval_st_data_warning_flow,
+  &ux_approval_st_flow_1_step,
+  &ux_approval_tx_data_warning_step,
+  &ux_approval_st_flow_2_step,
+  &ux_approval_from_address_step,
+  &ux_approval_confirm_step,
+  &ux_approval_reject_step
+);
+
+// TRANSFER
+//////////////////////////////////////////////////////////////////////
+UX_STEP_NOCB(
+    ux_approval_tx_1_step,
+    pnn,
+    {
+      &C_icon_certificate,
+      "Review",
+      "Transaction",
+    });
+UX_STEP_NOCB(
+    ux_approval_tx_2_step,
+    bnnn_paging,
+    {
+      .title = "Amount",
+      .text = (char *) G_io_apdu_buffer
```

Comment:
Doesn't seem great, but I guess not from you?

Reply:
@xchapron-ledger not from me at all. Legacy. Don't want to refacto the whole architecture...